rs/xnet/payload_builder/tests/certified_slice_pool.rs

```diff
@@ -1114,3 +1116,49 @@ proptest! {
         });
     }
 }
+
+proptest! {
+    #![proptest_config(ProptestConfig::with_cases(10))]
+
+    // Testing the 'signals limit' (the limit on messages in a slice such that the number of
+    // signals after inducting it is capped) requires streams with thousands of messages in it.
+    //
+    // It is therefore using a reduced number of cases to keep the load within reasonable bounds.
+    #[test]
+    fn pool_take_slice_respects_signal_limit(
```

Comment:
Made this test before I moved the change with `max_messages_end` down into `Payload::take_prefix()` so it's now perhaps not in the ideal place. But this tests what it should and if we make the `MAX_STREAM_MESSAGES` into an argument instead, I can probably integrate the test for the new limit as a 3 liner elsewhere.

I'd rather keep this as it is for now and think about the other change on a different PR. There is at least one issue with the detection of dishonest subnets due to too many signals. We can (and probably should) change that check to use 10k rather than 200k after this is has been rolled out for a while. But that assumes its a constant...